api/src/main/java/net/countercraft/movecraft/sign/AbstractMovecraftSign.java

```diff
@@ -0,0 +1,160 @@
+package net.countercraft.movecraft.sign;
+
+import net.countercraft.movecraft.craft.Craft;
+import net.countercraft.movecraft.craft.type.CraftType;
+import net.countercraft.movecraft.util.MathUtils;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer;
+import org.bukkit.entity.Player;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.block.SignChangeEvent;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.function.Function;
+
+// DONE: In 1.21 signs can have multiple sides! This requires us to pass the clicked side through or well the relevant lines and the set method for the clicked side => Resolved using the SignWrapper
+/*
+ * Base class for all signs
+ *
+ * A instance of a sign needs to be registered using the register function.
+ * Signs react to the following events:
+ *  - SignChangeEvent
+ *  - PlayerInteractEvent, if the clicked block is a sign
+ *  - CraftDetectEvent
+ *  - SignTranslateEvent (if the sign is a subclass of AbstractCraftSign)
+ *
+ *  Whenenver one of those events are cought by the AbstractSignListener instance, it is attempted to retrieve the relevant AbstractMovecraftSign instance.
+ *  For that, the first line of the sign's clicked side is extracted and formatting removed. If it matches the format "foo: bar", only "foo:" will be used.
+ *  With that ident, the sign is attempted to be retrieved vy tryGet(). If that returns something, the object's relevant method is called.
+ */
+public abstract class AbstractMovecraftSign {
+
+    private static final Map<String, AbstractMovecraftSign> SIGNS = Collections.synchronizedMap(new HashMap<>());
+
+    public static boolean hasBeenRegistered(final String ident) {
```

Comment:
After discussions with players on AP testing out 1.20, we should have support for aliases.  For example, Movecraft-Combat's "Cannon Director" sign does not fit on a hanging sign, so we should support also registering it to recognize "Cannon Dir".

Reply:
You could just register the same instance with a different name too, that's what i did in my implementations for the helm sign, as it reacts to [Helm] and the fancy thing. 

We could directly add optional aliases into the register function though which will then do the "additional" registering directly inside, which makes the "registering" part prettier